### PR TITLE
optimize CCD against btCompoundShapes with dynamicAabbTree

### PR DESCRIPTION
**Pull request by AndrewMeadows**
These are some changes we've been using for several months and are very happy with.  I'm finally getting around to submitting as a proper PR.

The motivation was to solve some performance problems we noticed when trying to do CCD against large btCompoundShapes.  We saw the **TODO** comment in the code about using the shape's "dynamic aabb tree" to speed up collision queries and decided to implement it.

### CODE REVIEW

File: src/BulletCollision/CollisionDispatch/btCollisionWorld.cpp [hunk 1]
```diff
@@ -790,23 +790,50 @@ void	btCollisionWorld::objectQuerySingleInternal(const btConvexShape* castShape,
 				}
 			}
 		} else {
-			///@todo : use AABB tree or other BVH acceleration structure!
```
Comment (AndrewMeadows) [inline src/BulletCollision/CollisionDispatch/btCollisionWorld.cpp]: Implementing this **TODO**.

**Comment by erwincoumans** [comment]
Great, let me have a look and merge it soon.
Thanks for contributing!

**Comment by erwincoumans** [comment]
Is there still some option to use the old (inefficient) path, for testing/comparing the new implementation?

**Comment by AndrewMeadows** [comment]
The new optimization is bypassed for `btCompoundShapes` that don't have an `m_dynamicAabbTree`, however the dynamic tree feature is enabled by default, which means any A/B example test under the current state of this PR would require some code changes and a recompile.

The other spots in the code that pivot their logic on the existence of `btCompoundShape::m_dynamicAabbTree` don't have global test configurations.  For example, the code in `btCompoundCollisionAlgorithm::processCollision()` which was, I think, the pattern followed when making this new optimization.

If we were to add a `btCollisionWorld::m_useCompoundDynamicAabbTreeForObjectQuery` data member it would be useful for evaluating this PR, but probably wouldn't be something wanted for the long term.  Who would use it once the optimized method is verified?  Nevertheless I'm willing to add it if it helps move things forward.  We could remove that test hack later once we're confident this is working correctly.

**Comment by erwincoumans** [comment]
We can have the optimization enabled by default, but add some option somewhere to disable it, just in case it causes issues. That way, we can ask people who have issues (unlikely) to try that option and see if it fixes their issue. Ideally we create a unit test comparing the two (I started with a few ones, and the build system supports automatically running them in CI)... Anyway, if you are too busy (which I understand) I may find some time next week to do this.

**Comment by erwincoumans** [comment]
I'll pull the request and write a test soon. Thanks for the contribution!

**Pull request merged**